Sources/Login/LoginCoordinator.swift

```diff
@@ -11,130 +11,101 @@ final class LoginCoordinator: NSObject,
+        self.isExpiredUser = isExpiredUser
     }
-    
+
     func start() {
-        let rootViewController = OnboardingViewControllerFactory
-            .createIntroViewController(analyticsService: analyticsCenter.analyticsService) { [unowned self] in
+        let rootViewController: UIViewController
+
+        if isExpiredUser {
+            let viewModel = OneLoginIntroViewModel(analyticsService: analyticsCenter.analyticsService) { [unowned self] in
                 authenticate()
             }
+            rootViewController = IntroViewController(viewModel: viewModel)
+        } else {
+            let viewModel = SignOutWarningViewModel(analyticsService: analyticsCenter.analyticsService) { [unowned self] in
+                authenticate()
+            }
+            rootViewController = GDSErrorViewController(viewModel: viewModel)
+        }
+
         root.setViewControllers([rootViewController], animated: true)
-        introViewController = rootViewController
-        checkLocalAuth()
-        showLoginErrorIfNecessary()
+
         launchOnboardingCoordinator()
-        NotificationCenter.default
-            .addObserver(self,
-                         selector: #selector(returnToIntroScreen),
-                         name: Notification.Name(.returnToIntroScreen),
-                         object: nil)
-    }
-    
-    private func checkLocalAuth() {
-        guard loginError as? PersistentSessionError == .userRemovedLocalAuth else {
-            return
-        }
-        let signOutWarningScreen = ErrorPresenter
-            .createSignOutWarning(analyticsService: analyticsCenter.analyticsService) {
-                self.root.dismiss(animated: true)
-            }
-        signOutWarningScreen.modalPresentationStyle = .overFullScreen
-        root.present(signOutWarningScreen, animated: false)
     }
-    
-    private func showLoginErrorIfNecessary() {
-        if loginError as? TokenError == .expired {
-            let signOutWarningScreen = ErrorPresenter
-                .createSignOutWarning(analyticsService: analyticsCenter.analyticsService) { [unowned self] in
-                    authenticate { [unowned self] in
-                        root.dismiss(animated: true)
-                    }
-                }
-            signOutWarningScreen.modalPresentationStyle = .overFullScreen
-            root.present(signOutWarningScreen, animated: false)
-        } else if let loginError {
-            let unableToLoginErrorScreen = ErrorPresenter
-                .createUnableToLoginError(errorDescription: loginError.localizedDescription,
-                                          analyticsService: analyticsCenter.analyticsService) { [unowned self] in
-                    root.dismiss(animated: true)
+
+    func authenticate() {
+//        guard sessionManager.isReauthSupported else {
+//            // DELETE DATA
+//            NotificationCenter.default.post(name: .clearWallet, object: nil)
+//            return
+//        }
```

Comment:
Guess we can get rid of this once we have the guard condition at the start of `startSession` confirmed

Reply:
Yes, I've removed now 🙂